lib/message_response.js

```diff
@@ -39,10 +40,15 @@ Response.prototype._validate = function () {
   return true;
 };
 
-Response.prototype.isValid = function () {
+// TODO: should probably be renamed isValidForSync
+Response.prototype.isValidForSync = function () {
```

Comment:
the difference in meaning between isValid and isValidForSync should be documented (it's not clear to me what this method does)